### PR TITLE
Webpack & Node tooling upgrades

### PR DESCRIPTION
**Pull request by thibaudcolas**
This PR was originally about updating our development tools to reduce the number of vulnerabilities we may be exposed to, as a follow-up to #6101. Now it also feels like an important next step from #6472.

Here are the changes:

- Upgrading to Webpack 4 to fix security vulnerabilities, and so we can eventually use `ts-loader` with TypeScript v4 for JS & TS compilation.
- Switching from `gulp-autoprefixer` and `gulp-cssnano` to `gulp-postcss`, and the standalone `cssnano` and `autoprefixer` pacakges. This means we can use the latest versions of those packages, and it’s also taking us closer to moving stylesheets compilation to Webpack eventually (working on this in a separate branch).
- A few other package upgrades to use versions with no known vulnerabilities.

There is now only 1 reported vulnerability in a transitive dependency of the Draftail editor, for which I’m certain we don’t use any of the vulnerable code.

---

* [x] Do the tests still pass? (https://docs.wagtail.io/en/latest/contributing/developing.html#testing)
* [x] Does the code comply with the style guide? (Run `make lint` from the Wagtail root)
* [x] For Python changes: Have you added tests to cover the new/fixed behaviour?
* [x] For front-end changes: Did you test on all of Wagtail’s supported browsers?  Chrome and IE11.
* [x] For new features: Has the documentation been updated accordingly?

### CODE REVIEW

File: client/webpack/dev.config.js [hunk 1]
```diff
@@ -1,23 +0,0 @@
...
-
-// add poll-options for in vagrant development
-// See http://andrewhfarmer.com/webpack-watch-in-vagrant-docker/
-config.watchOptions = {
-  poll: 1000,
-  aggregateTimeout: 300,
-};
-
-// Set process.env.NODE_ENV to development to enable JS development aids.
-config.plugins.push(new webpack.DefinePlugin({
```
Comment (kaedroho) [inline client/webpack/dev.config.js:17]: What about these plugins?

File: client/webpack/prod.config.js [hunk 2]
```diff
@@ -1,30 +0,0 @@
-var webpack = require('webpack');
-var base = require('./base.config');
-var config = base('production');
-
-
-// production overrides go here
-config.plugins.push(new webpack.DefinePlugin({
```
Comment (kaedroho) [inline client/webpack/prod.config.js:7]: What about these plugins?

**Review by kaedroho — COMMENTED** [review]
Changes look good! Wondering what we're doing about the plugins that were removed in the second commit?

I've just done some squashing of the commits in: https://github.com/kaedroho/wagtail/commits/chore/major-npm-updates

File: client/webpack/dev.config.js (hunk 1 above)
Comment (thibaudcolas) [reply client/webpack/dev.config.js:17]: This is automatically defined in Webpack 4 based on the `mode`.

**Review by kaedroho — APPROVED** [review]

**Pull request merged**